OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -164,61 +181,30 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Joint, ModelComponent);
     const std::string& getChildFrameName() const;
 
     /**
-     * Set the child body that this joint connects.
-     *
-     * @param child PhysicalFrame reference.
-     */
-    void setChildFrame(const PhysicalFrame& child);
-
-    /**
      * Get the child joint frame.
      *
      * @return const PhysicalFrame reference.
      */
     const PhysicalFrame& getChildFrame() const;
 
-    void setLocationInChild(const SimTK::Vec3& aLocation);
-    const SimTK::Vec3& getLocationInChild() const;
-    void setOrientationInChild(const SimTK::Vec3& aOrientation);
-    const SimTK::Vec3& getOrientationInChild() const;
-
     // Relating to the parent joint frame
     void setParentFrameName(const std::string& aName);
     const std::string& getParentFrameName() const;
 
     /**
-    * Set the parent frame that this joint connects.
-    *
-    * @param parent PhysicalFrame.
-    */
-    void setParentFrame(const PhysicalFrame& parent);
-    /**
      * Get the parent frame to which this joint attaches.
      *
      * @return const ref to parent PhysicalFrame.
      */
     const OpenSim::PhysicalFrame& getParentFrame() const;
 
-    void setLocationInParent(const SimTK::Vec3& aLocation);
-    const SimTK::Vec3& getLocationInParent() const;
-    void setOrientationInParent(const SimTK::Vec3& aOrientation);
-    const SimTK::Vec3& getOrientationInParent() const;
-
-    /** Get the Joint frames expressed as body transforms. Only available after
-        extendConnectToModel() has been called on the Joint. */
-    const SimTK::Transform& getParentTransform() const
-        { return _jointFrameInParent; }
-    const SimTK::Transform& getChildTransform() const
-        { return _jointFrameInChild; }
-
```

Comment:
What should people do if they were using these methods? Two suggestions:

1. still implement these methods, have them throw an exception if called (like OpenSim::DeprecatedException, or OpenSim::NoLongerExists) and also add a DEPRECATED_14.
2. Add comments to the CHANGELOG explaining what people should do instead to achieve the same functionality.